dirty_cat/datasets/tests/test_fetching.py

```diff
@@ -1,238 +1,352 @@
-                    encoding='latin-1'),),
-                main_file='unzipped_data.txt',
-                source='http://foo/')
-
-            info_unzipped = fetching.fetch_dataset(
-                dataset_info_with_zipfile, show_progress=False)
-
-            with open(info_unzipped['path'], 'rb') as f:
-                content_unzipped = f.read()
-
-            assert content_unzipped == b'\xc3\xa9'
+            # Valid call
+            returned_info = fetch_openml_dataset(dataset_id=test_dataset["id"],
+                                                 data_directory=test_data_dir)
+
+        except URLError:
+            # No internet connection, or the website is down.
+            # We abort the test.
+            #
+            # One could try to manually recreate the tree structure
+            # created by ``fetch_openml()``,  and the
+            # ``.gz`` files within in order to finish the test.
+            return
+
+        assert returned_info["description"].startswith(test_dataset["desc_start"])
+        assert returned_info["source"] == test_dataset["url"]
+        assert returned_info["path"] == test_data_dir / test_dataset["csv_name"]
+
+        assert returned_info["path"].is_file()
+
+        dataset: pd.DataFrame = pd.read_csv(returned_info["path"], sep=",",
+                                            quotechar="'", escapechar="\\")
+
+        assert dataset.shape == (test_dataset["dataset_rows_count"],
+                                 test_dataset["dataset_columns_count"])
 
     finally:
-        if os.path.exists(datadir):
-            shutil.rmtree(datadir)
+        shutil.rmtree(path=str(test_data_dir), ignore_errors=True)
+
+
+@mock.patch("pathlib.Path.is_file")
+@mock.patch("dirty_cat.datasets.fetching._get_features")
+@mock.patch("dirty_cat.datasets.fetching._get_details")
+@mock.patch("dirty_cat.datasets.fetching._export_gz_data_to_csv")
+@mock.patch("dirty_cat.datasets.fetching._download_and_write_openml_dataset")
+def test_fetch_openml_dataset_mocked(mock_download, mock_export,
+                                     mock_get_details, mock_get_features,
+                                     mock_pathlib_path_isfile):
+    """
+    Test function ``fetch_openml_dataset()``, but this time,
+    we mock the functions to test its inner mechanisms.
+    """
+
+    from dirty_cat.datasets.fetching import fetch_openml_dataset, Details, Features
+
+    mock_get_details.return_value = Details("Dataset_name", "123456",
+                                            "Dummy dataset description.")
+    mock_get_features.return_value = Features(["id", "name", "transaction_id",
+                                               "owner", "recipient"])
+
+    test_data_dir = get_test_data_dir()
+
+    # We test the function to see if it behaves correctly when files exists
+    mock_pathlib_path_isfile.return_value = True
+
+    fetch_openml_dataset(50, test_data_dir)
+
+    mock_download       .assert_not_called()
```

Comment:
Strange formatting. I don't think that this is PEP8 (but correct me if I'm wrong)